pkg/registry/apis/provisioning/import.go

```diff
@@ -172,41 +176,29 @@ func (c *importConnector) Connect(
 				}
 			}
 
-			wrapper := file.AsResourceWrapper()
-			resource := wrapper.Resource.File
-			resource.SetNestedField(name, "metadata", "name")
-			resource.SetNestedField(ns, "metadata", "namespace")
-			if folder := path.Base(dir); folder != "." && folder != "/" {
-				resource.SetNestedField(path.Base(dir), "metadata", "annotations", apiutils.AnnoKeyFolder)
+			if file.gvr == nil {
+				logger.DebugContext(ctx, "no GVR found")
+				continue
 			}
 
-			gv, err := schema.ParseGroupVersion(resource.GetNestedString("apiVersion"))
-			if err != nil {
-				logger.DebugContext(ctx, "invalid GroupVersion was given as the apiVersion of the parsed object?", "error", err)
-				responder.Error(apierrors.NewInternalError(fmt.Errorf("got an invalid groupversion from a parsed file %s: %w", entry.Path, err)))
-				return
-			}
-			gvk := gv.WithKind(resource.GetNestedString("kind"))
-			logger = logger.With("gvk", gvk)
-			gvr, ok := kinds.Resource(gvk)
-			if !ok {
-				logger.DebugContext(ctx, "got GVK of a resource we don't know how to control; ignoring")
-				continue
+			file.obj.SetName(name)
+			if folder := path.Base(dir); folder != "." && folder != "/" {
+				file.meta.SetFolder(path.Base(dir))
 			}
-			iface := client.Resource(gvr).Namespace(ns)
```

Comment:
the parser already created a client!

🤔  we likely need a better name for that since it:

1. parses the file
2. finds a GVK/GVR
3. when 'validate' is true, it then runs DryRun

Reply:
I only remembered about this after seeing it in my refactor yesterday 😄 
I agree a better name might be prudent, but I also can't come up with one